### PR TITLE
Deprecate WeekSubscribe method due to high probability of misuse

### PR DESCRIPTION
**Pull request by maxkoshevoi**
### Description of Change ###

As discussed in #1001, deleted `WeekSubscribe` since it's most likely will be misused. 

~~It's not released yet, so this is not a breaking change.~~
This method is now released in 1.1.0. Maybe it's deletion should be reevaluated.

### Bugs Fixed ###

- Fixes #1001

### API Changes ###

Removed:
- `void WeakSubscribe<T>(this INotifyPropertyChanged target, T subscriber, Action<T, object?, PropertyChangedEventArgs> action)` 
 
### Behavioral Changes ###

None

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard

- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Comment by brminnick** [comment]
We would need to first mark it as `[Obsolete]` in the next release.

Then we could remove it in the next major release.

**Comment by pictos** [comment]
@brminnick oh yeah, We already released the 1.1 in a stable channel! I forgot it. @maxkoshevoi please restore that file and add the `Obsolete` attribute as suggested by @brminnick, thanks.

**Comment by maxkoshevoi** [comment]
@pictos Done

File: src/CommunityToolkit/Xamarin.CommunityToolkit/ObjectModel/Extensions/INotifyPropertyChangedExtension.shared.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@ namespace Xamarin.CommunityToolkit.ObjectModel.Extensions
 {
 	public static class INotifyPropertyChangedExtension
 	{
+		[Obsolete("This method shouldn't be used week events (like one in ObservableObject). This method is deprecated due to high probability of misuse.")]
```
Comment (pictos) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/ObjectModel/Extensions/INotifyPropertyChangedExtension.shared.cs:8]:
```suggestion
        [EditorBrowsable(EditorBrowsableState.Never)]
		[Obsolete("This method shouldn't be used week events (like one in ObservableObject). This method is deprecated due to high probability of misuse.")]
```

**Comment by pictos** [comment]
Thanks, @maxkoshevoi, now we just need to fix the CI to merge this puppy (:

**Comment by maxkoshevoi** [comment]
@pictos This is not related to localization. `WeekSubscribe` to subscribe to any property change event.

**Comment by pictos** [comment]
> @pictos This is not related to localization. `WeekSubscribe` to subscribe to any property change event.

but this isn't created because of `LocalizedString` feature? I didn't found any better label for that

**Comment by maxkoshevoi** [comment]
> but this isn't created because of `LocalizedString` feature? I didn't find any better label for that

Yes, but it's still a general purpose method. 
Yeah, there isn't really a label for it 😄 What about `Breaking change`? Most users have `Treat warnings as errors set`, so this will be a breaking change for them.

**Review by pictos — APPROVED** [review]

**Pull request merged**